src/Tasks/Microsoft.NET.Build.Tasks/GenerateShims.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using NuGet.Frameworks;
+using NuGet.ProjectModel;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public sealed class GenerateShims : TaskBase
+    {
+        private NuGetPackageResolver _packageResolver;
+
+        /// <summary>
+        /// Path to assets.json.
+        /// </summary>
+        public string ProjectAssetsFile { get; set; }
+
+        /// <summary>
+        /// The file name of Apphost asset.
+        /// </summary>
+        [Required]
+        public string DotNetAppHostExecutableNameWithoutExtension { get; set; }
+
+        /// <summary>
+        /// Path to project file (.csproj|.vbproj|.fsproj)
+        /// </summary>
+        [Required]
+        public string ProjectPath { get; set; }
+
+        /// <summary>
+        /// TFM to use for compile-time assets.
+        /// </summary>
+        [Required]
+        public string TargetFrameworkMoniker { get; set; }
+
+        /// <summary>
+        /// PackageId of the dotnet tool NuGet Package.
+        /// </summary>
+        [Required]
+        public string PackageId { get; set; }
+
+        /// <summary>
+        /// Package Version of the dotnet tool NuGet Package.
+        /// </summary>
+        [Required]
+        public string PackageVersion { get; set; }
+
+        /// <summary>
+        /// The command name of the dotnet tool.
+        /// </summary>
+        [Required]
+        public string ToolCommandName { get; set; }
+
+        /// <summary>
+        /// The entry point of the dotnet tool which will be run by Apphost
+        /// </summary>
+        [Required]
+        public string ToolEntryPoint { get; set; }
+
+        /// <summary>
+        /// The output Directroy path of generated shims.
+        /// </summary>
+        [Required]
+        public string PackagedShimOutputDirectory { get; set; }
+
+        /// <summary>
+        /// The RuntimeIdentifiers that shims will be generated for.
+        /// </summary>
+        [Required]
+        public ITaskItem[] PackageToolShimRuntimeIdentifiers { get; set; }
```

Comment:
>PackageToolShimRuntimeIdentifiers [](start = 27, length = 33)

I think we should rename this to match the property, or shorten it and decide there's enough context here to just "ShimRuntimeIdentifiers" or even "RuntimeIdentifiers". Right now, the unnecessary subtle difference between "PackAsToolShimRuntimeIdentifiers" and "PackageToolShimRuntimeIdentifiers" is confusing me.

Reply:
Fixed, changed to ShimRuntimeIdentifiers